service/src/main/java/bio/terra/externalcreds/services/PassportService.java

```diff
@@ -91,34 +97,62 @@ public ValidatePassportResultInternal validatePassport(
           if (visaComparator.visaTypeSupported(visa)
               && visa.getIssuer().equals(criterion.getIssuer())
               && visaComparator.matchesCriterion(visa, criterion)) {
-            var linkedAccount =
-                linkedAccountsByJwtId.get(passportWithVisas.getPassport().getJwtId());
             var auditInfoMap =
-                new HashMap<>(
-                    Map.of(
-                        "passport_jti",
-                        passportWithVisas.getPassport().getJwtId(),
-                        "external_user_id",
-                        linkedAccount.getExternalUserId(),
-                        "internal_user_id",
-                        linkedAccount.getUserId()));
+                new HashMap<>(Map.of("passport_jti", passportWithVisas.getPassport().getJwtId()));
             transactionClaim.map(t -> auditInfoMap.put("txn", t));
-            return new ValidatePassportResultInternal.Builder()
-                .valid(true)
-                .matchedCriterion(criterion)
-                .auditInfo(auditInfoMap)
-                .build();
+            var jwtId = passportWithVisas.getPassport().getJwtId();
+            var linkedAccount =
+                Optional.ofNullable(
+                    linkedAccountsByJwtId.get(passportWithVisas.getPassport().getJwtId()));
+            var passportValid =
+                linkedAccount
+                    .map(
+                        l -> {
+                          log.info("Found internal user {} for passport {}", l.getUserId(), jwtId);
+                          auditInfoMap.putAll(
+                              Map.of(
+                                  "external_user_id", l.getExternalUserId(),
+                                  "internal_user_id", l.getUserId()));
+                          return true;
+                        })
+                    .orElseGet(
+                        () -> {
+                          log.info(
+                              "No internal user found for passport {}. Checking issue time...",
+                              passportWithVisas.getPassport().getJwtId());
+                          return isPassportIssueTimeValid(passportWithVisas.getPassport());
+                        });
```

Comment:
this function feels like it's getting a bit unwieldy. could we pull out this logic into a helper function?